packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -695,136 +700,317 @@ class _SortData {
+///   const DemoButton({this.name, this.autofocus = false, this.order});
+///
+///   final String name;
+///   final bool autofocus;
+///   final double order;
+///
+///   void _handleOnPressed() {
+///     print('Button $name pressed.');
+///     debugDumpFocusTree();
+///   }
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return FocusTraversalOrder(
+///       order: NumericFocusOrder(order),
+///       child: FlatButton(
+///         autofocus: autofocus,
+///         focusColor: Colors.red,
+///         onPressed: () => _handleOnPressed(),
+///         child: Text(name),
+///       ),
+///     );
+///   }
+/// }
+/// ```
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return DefaultFocusTraversal(
+///     policy: OrderedFocusTraversalPolicy(),
+///     child: Column(
+///       mainAxisAlignment: MainAxisAlignment.center,
+///       children: <Widget>[
+///         Row(
+///           mainAxisAlignment: MainAxisAlignment.center,
+///           children: const <Widget>[
+///             DemoButton(name: 'Six', order: 6),
+///           ],
+///         ),
+///         Row(
+///           mainAxisAlignment: MainAxisAlignment.center,
+///           children: const <Widget>[
+///             DemoButton(name: 'Five', order: 5),
+///             DemoButton(name: 'Four', order: 4),
+///           ],
+///         ),
+///         Row(
+///           mainAxisAlignment: MainAxisAlignment.center,
+///           children: const <Widget>[
+///             DemoButton(name: 'Three', order: 3),
+///             DemoButton(name: 'Two', order: 2),
+///             DemoButton(name: 'One', order: 1, autofocus: true),
+///           ],
+///         ),
+///       ],
+///     ),
+///   );
+/// }
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [NumericFocusOrder], a focus order that assigns a numeric traversal order
+///    to a [FocusTraversalOrder] widget.
+///  * [LexicalFocusOrder], a focus order that assigns a string-based lexical
+///    traversal order to a [FocusTraversalOrder] widget.
+///  * [FocusOrder], an abstract base class for all types of focus traversal
+///    orderings.
+class OrderedFocusTraversalPolicy extends FocusTraversalPolicy with DirectionalFocusTraversalPolicyMixin {
+  /// Constructs a traversal policy that orders widgets for keyboard traversal
+  /// based on an explicit order.
+  ///
+  /// If [secondary] is null, it will default to [ReadingOrderTraversalPolicy].
+  OrderedFocusTraversalPolicy({this.secondary}) {
+    secondary ??= ReadingOrderTraversalPolicy();
+  }
+
+  /// This is the policy that is used when a node doesn't have an order
+  /// assigned, or when multiple nodes have orders which are identical.
+  ///
+  /// If not set, this defaults to [ReadingOrderTraversalPolicy].
+  ///
+  /// This policy determines the secondary sorting order of nodes which evaluate
+  /// as having an identical order (including those with no order specified).
```

Comment:
Isn't it must likely a bug if I have two widgets with `NumericFocusOrder(1.0)`? This fallback seems to be hiding that?